src/libponyc/type/matchtype.h

```diff
@@ -44,6 +44,14 @@ typedef enum
 matchtype_t is_matchtype(ast_t* operand, ast_t* pattern, errorframe_t* errorf,
   pass_opt_t* opt);
 
+/**
+ * Same as is_matchtype() but specifically designed to work with `as` which
+ * under the steed model, needs to act differently than matchwith respect
+ * to iso.
+ */
+matchtype_t is_astype(ast_t* operand, ast_t* pattern, errorframe_t* errorf,
```

Comment:
this will be is_matchtype

current is_matchtype will be is_matchtype_with_consumed_pattern